File: code/src/main/python/misconceptions/pdUtils/snippet_runner.py
import sys
import os

sys.path.append(os.path.abspath("."))
sys.dont_write_bytecode = True

__author__ = "bigfatnoob"

import pandas as pd
import numpy

INPUT_PATH = "/Users/panzer/Raise/ProgramRepair/CodeSeer/code/src/main/python/misconceptions/resources/titanic.csv"
OUTPUT_PATH = "/Users/panzer/Raise/ProgramRepair/CodeSeer/code/results/py_executed.csv"


def to_csv(obj):
  if type(obj).__module__ == "numpy":
    pd.DataFrame(obj).to_csv(OUTPUT_PATH)
  else:
    obj.to_csv(OUTPUT_PATH)

def evaluate(snippet):
  df = pd.read_csv(INPUT_PATH)[:10]
  res = eval(snippet)
  to_csv(res)


# evaluate("df.describe()")
# evaluate("df.Survived.values")
# evaluate("df.tail()")
# evaluate("df['Cabin'].apply(lambda x: str(x)[0])")
# evaluate("df.groupby('Ticket')['Name'].transform('count')")
# evaluate("numpy.mean(df, axis=0)")
# evaluate("df.Age.describe()")
# evaluate("df.as_matrix(columns=['Survived'])")
evaluate("df['Sex'].value_counts(normalize=True)")

File: code/src/main/r/parser/snippet_runner.R
# Title     : TODO
# Objective : TODO
# Created by: panzer
# Created on: 6/6/19

library(base)

INPUT_PATH = "/Users/panzer/Raise/ProgramRepair/CodeSeer/code/src/main/python/misconceptions/resources/titanic.csv"
OUTPUT_PATH = "/Users/panzer/Raise/ProgramRepair/CodeSeer/code/results/r_executed.csv"

executor <- function(snippet) {
    df = head(read.csv(INPUT_PATH), 10)
    res = eval(parse(text=snippet))
    # print(res)
    write.csv(res, OUTPUT_PATH)
}

# executor("summary(df)")
# executor("df$Survived")
# executor("df[,]")
# executor("substr(df$Cabin, 1, 1)")
# executor("ifelse(df$Survived > 0.5, 1, 0)")
# executor("table(df$Ticket)[df$Ticket]")
# executor("mean(df)")
# executor("df$Age")
# executor("prop.table(table(df$Survived))")
# executor("levels(df$Sex)")

train = read.csv(INPUT_PATH)
# print(train[is.na(train$Embarked)==TRUE, ])
print(train[train$Embarked == NA ])
# print(train[na.omit(train$Embarked !=NA),])